### ISSUE TITLE
Support for Mocks + other features?

### ISSUE THREAD

**Issue by stefano-p** [jasmcaus/tau#8]
Hello, do you plan to include support for Mocks? What do you think is still needed or has to be improved?

**Comment by jasmcaus** [comment]
Personally, I haven't used Mocks. That's why Tau doesn't have it yet. Feel free to experiment with them and maybe open a PR and we can decide later.

**Comment by stefano-p** [comment]
Ok, so as soon as I have a bit of time I can try something and once I am happy with it, I will submint a PR.

**Comment by jasmcaus** [comment]
Sure. Feel free to submit a PR when ready

**Comment by kfrolov** [comment]
I think, for such small test framework you need small mocking framework, like this: https://github.com/ztbrown/Simulacrum
Pair of Tau and Simulacrum looks promisingly. Despite both may have many issues, at least it's easy to fix both.

**Comment by matu3ba** [comment]
If you base your design of the code base around mocks, then you have lost already.
Mocking is a [quick hack](https://www.accenture.com/us-en/blogs/software-engineering-blog/to-mock-or-not-to-mock-is-that-even-a-question) and incurs long-time costs on your code base.

Every code base must be inspectable more or less effortlessly and if you cant do that, then you already broke weak coupling preventing or adding heavy costs on refactorings: You cant specify the state of things via tests as invariants, because it is specified via state of other stuff (the mock).

Refactoring with explicit function signatures of tests is simple, as you can use regexes. You cant do this with mocking techniques.

**Comment by jasmcaus** [comment]
I agree. I wasn't too sure if we needed to have Mocks in the first place and whether it was worth the headache. Probably not, but I'm open to ideas (and contributors too haha)
